### ISSUE TITLE
Spaceflight tutorial feedback

### ISSUE THREAD

**Issue by picklejuicedev** [kedro-org/kedro#2003]
## Description
As I am new to Kedro I wanted to help out with improving the documentation. I ran through the current released version of the SpaceFlight tutorial last week (26/10/2022) and now did the same again with the updated version from the branch `kedro-1874-spaceflight-improvements` with all the updates until today (4/11/2022).

## Context
Being new to Kedro I remembered the stumbling points when using Kedro for the first time and it will be useful feedback to improve the documentation.

## Possible Implementation
I forked the current repo and made changes to the documentation as suggestions. Find them [here](https://github.com/kedro-org/kedro/compare/kedro-1874-spaceflights-improvements...picklejuicedev:kedro:kedro-1874-spaceflights-improvements)

Some notes:
- Generally I had no issues following the tutorial, it is now much clearer what code to paste where.
- I removed a bunch of changes that Ja had just put in (sorry) to clarify what you don't need to do when using the starter. I used the starter first time round and it didn't matter that the comments weren't there. I just saw the code was already there and then reviewed it. With this being mentioned every time it disrupts the flow somewhat, so I changed it to have a {note} at the top of each chapter. I thought this works a bit better.
- Data science pipeline: I had a go at improving the modular pipeline section. I totatlly didn't get this at first with how namespaces affect paramters and how you can "freeze" them. See what you think - is a tricky topic.
- install.md: still referenced discord server
- I stopped at "Package a Kedro project", may have a look at that next, but please give me feedback to what I have done to date and how to improve process.

## Further suggestions
Here are some other observations that I didn't include in the repo, but you may want to consider:

### Installation prerequisites - python version
The installation suggests to use conda with a python environment of 3.8.
It had me baffled as to why Kedro prefers that version. I ignored it and used 3.10 and it works fine. I later saw that in your repo it also flags newer versions as supported. Maybe this needs clarifying?

### First page - Kedro spaceflight tutorial - remove git section
Git is of course a critical tool and workflow for any developer. But I was surprised to find this section here and after following it (do I really need a branch `feature/project-template` ?) only to find out that it isn't needed anywhere else in the tutorial. I  would suggest to move this to a separate section like "git best practice" and also talk about jupyter notebook management and how to manage data correctly as it shouldn't be in a git repo (as it is with the starter). So all a bit confusing and do we really need this here?

### Data Science - modular pipeline
Here is one that I didnt' change but still find confusing from an architectural level:

Earlier in the tutorial we use:
```
from kedro.pipeline.modular_pipeline import pipeline
```
and in modular pipelines we use:
```
from kedro.pipeline import Pipeline, node, pipeline
```

2 pipelines with the exact same name doing different things? Mind blown. (I am sure there is a good explanation...)

### Warning in output
There is a DeprecationWarning in the output when using the tutorial about `distutils`. Maybe some dependency needs updating?
You can find it in the final output of the data-science pipeline:
```
[11/02/22 10:41:06] WARNING  /Users/jo_stichbury/opt/anaconda3/envs/py38/lib/python3.8/site-packages/plotly/graph_objects/ warnings.py:109
                             __init__.py:288: DeprecationWarning: distutils Version classes are deprecated. Use
                             packaging.version instead.
                               if LooseVersion(ipywidgets.__version__) >= LooseVersion("7.0.0"):
```

**Comment by stichbury** [comment]
Thank you @picklejuicedev that is really helpful and I agree with you about removing all those notes about not copying code if you're using the starter. I'm on McK onboarding this week so won't be doing much with the docs, but I'll pick it up as soon as I can and merge in with the main PR. Hope to see this all go live in November.

Thanks again -- I really appreciate it. I've been updating the Get Started info too, but there's no PR out as yet. When I have one, I'll tag you on it in case you want to review. 

🙏

**Comment by picklejuicedev** [comment]
Hi @stichbury - glad to be of help and let me know if there are any questions.
I had a play with the last section (Package a Kedro Project) today and no changes to any md files, just notes here:

## Documentation

Building the documentation works as described. `kedro build-docs --open` will install any requirements, build your documentation and then open in the browser. But I was very confused when I looked at it. It seems to have a complicated hierarchy with many links pointing to the same pages. So by default there are just 2 pages useful: those for the 2 submodules == pipelines generated. Sorry, I was quite underwhelmed by this feature. It is a shame as I think it is a great idea - building sustainable software is all about being able to explain and share and a documentation template / framework is definitely on my list ;)

I saw that this feature will be discontinued with 0.19 which makes sense as in its current state it is pretty tricky to use - you still need to learn Sphinx to do anything, so you may as well just use that. Maybe at some point in the future this can be revisited? Possibly as an external tool - doesn't need to be in the core product I guess.

I would suggest putting a Deprecation note into the spaceflight tutorial or remove the section completely. From the perspective of a new user (me) I didn't see the message in the logs as it flies by pushed away by the dependency installations.

## Package your project

Worked as described first time and I could pip install the wheel and use it in a new virtual python envinonment (using Windows and poetry as envionment, al on python 3.10.7).

**Comment by stichbury** [comment]
Thanks for all this useful feedback. I've implemented changes in the PR to address #1874 and am mindful of any that haven't been introduced (yet) such as 

* feedback about `build-docs` (which I've not deprecated as it's still something I'm considering -- we aren't removing Sphinx functionality, just removing the command and relying on usage of direct Sphinx commands, which could be optimistic).
* feedback about `git` which I'm tackling in a separate PR

I'll close this issue now @picklejuicedev 🙏

**Comment by picklejuicedev** [comment]
@stichbury - I am happy my suggestions were helpful and I totally understand that some items will take some time to consider. Ping me if you need someone to look over anything else with fresh eyes.
